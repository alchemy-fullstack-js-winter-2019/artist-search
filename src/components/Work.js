import React from 'react';
import PropTypes from 'prop-types';
import { Link, BrowserRouter as Router } from 'react-router-dom';
import styles from './css/Work.css';

export default function Work({ artist, work }){
  return (
    <>
    <Router>
      <Link to={`/work/${artist}/${work.title}`} key={work.id} className={styles.work}>
        <li key={work.id}>{work.title}</li>
      </Link>
    </Router>
    </>
  );
}

Work.propTypes = {
  work: PropTypes.object.isRequired,
  artist: PropTypes.string.isRequired
};
