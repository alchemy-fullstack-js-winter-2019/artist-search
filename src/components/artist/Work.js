import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';




function ArtistWork({ values }) {
    const { work, name } = values;

    return (
        <Link to={`/${name}/${work}`}>
            <h2> {work}</h2>
            <h2> {name}</h2>
        </Link>
    );
}
ArtistWork.propTypes = {
    values: PropTypes.object,
};

export default ArtistWork;
