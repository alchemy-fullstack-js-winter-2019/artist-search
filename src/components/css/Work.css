.work {
  font-family: 'Lucida Sans', 'Lucida Sans Regular', 'Lucida Grande', 'Lucida Sans Unicode', Geneva, Verdana, sans-serif;
  list-style: none;
  text-decoration: none;
  text-align: center;
  padding: 5px;
  :hover {
    background: rgba(255, 243, 76, 0.781);
    padding: 200px;
  }
}
