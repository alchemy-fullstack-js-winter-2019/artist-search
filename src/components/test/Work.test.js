import React from 'react';
import Work from '../Work';
import renderer from 'react-test-renderer';

describe('Work', () => {
  it('matches a snapshot', () => {
    const artist = 'Mumford and Sons';
    const work =     {
      'type-id': null,
      'languages': [
        'eng'
      ],
      'iswcs': [],
      'id': '15dc1e5d-7aff-488e-a6ff-40ba51f326b5',
      'disambiguation': '',
      'attributes': [],
      'type': null,
      'language': 'eng',
      'title': 'Beloved'
    };
    const tree = renderer.create(
      <Work artist={artist} work={work} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
