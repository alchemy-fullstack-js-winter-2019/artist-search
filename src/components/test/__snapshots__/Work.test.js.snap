// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Work matches a snapshot 1`] = `
<a
  className="work"
  href="/work/Mumford and Sons/Beloved"
  onClick={[Function]}
>
  <li>
    Beloved
  </li>
</a>
`;
